src/Fixer/FunctionNotation/FopenFlagsFixer.php

```diff
@@ -21,21 +24,42 @@
 /**
  * @author SpacePossum
  */
-final class FopenFlagsFixer extends AbstractFopenFlagFixer
+final class FopenFlagsFixer extends AbstractFopenFlagFixer implements ConfigurationDefinitionFixerInterface
 {
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+    }
+
     /**
      * {@inheritdoc}
      */
     public function getDefinition()
     {
         return new FixerDefinition(
-            'The flags in `fopen` calls must contain `b` and must omit `t`.',
+            'The flags in `fopen` calls must omit `t`, and `b` must be omitted or included consistently.',
             [new CodeSample("<?php\n\$a = fopen(\$foo, 'rwt');\n")],
             null,
             'Risky when the function `fopen` is overridden.'
         );
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('b_mode', 'The `b` flag must be used (`true`) or omitted (`false`).'))
+                ->setAllowedTypes(['bool'])
+                ->setDefault(true)
```

Comment:
what is expected behaviour under 3.0 ?

Reply:
same i guess, or do you want me to change it on 3.0?

Reply:
I think `false`, as mentioned in #4011 it's the default for all supported PHP versions